updateFeatureFlag/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/Real-Dev-Squad/feature-flag-backend/database"
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/Real-Dev-Squad/feature-flag-backend/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/go-playground/validator/v10"
+	"log"
+	"net/http"
+	"time"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+}
+
+func handleValidationError(err error) []utils.ValidationError {
+	var errors []utils.ValidationError
+
+	for _, err := range err.(validator.ValidationErrors) {
+		errors = append(errors, utils.ValidationError{
+			Field: err.Field(),
+			Error: err.Tag(),
+		})
+		log.Println("Errors are ", errors)
+	}
+	return errors
+}
+
+// creating instance of databae
+func updateFeatureFlag(featureFlag *models.FeatureFlag) (events.APIGatewayProxyResponse, error) {
+	db := database.CreateDynamoDB()
+
+	item, err := dynamodbattribute.MarshalMap(featureFlag)
+	if err != nil {
+		return utils.ServerError(err)
+	}
+
+	input := &dynamodb.PutItemInput{
+		Item:      item,
+		TableName: aws.String(database.GetFeatureFlagTableName()),
+	}
+
+	_, err = db.PutItem(input)
```

Comment:
Is there a need to check if the feature flag id is valid by fetching the future flag details?

Reply:
I think we can use the UpdateItem function directly with a conditional expression `attribute_exists` and passing the id.
So it will give an error if an item with the given id does not exist and then we can return a 404 status code.
It will save us the extra GetItem query

Reference - 
https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/expression/#AttributeExists
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ConditionExpressions.html#Expressions.ConditionExpressions.SimpleComparisons

Reply:
Will try implementing this.

Reply:
This is not yet resolved, will push the changes tomorrow morning.

Reply:
Pushed the changes @shubham-y have a look

Reply:
@shubham-y gentle reminder